nixos/modules/services/web-apps/pds.nix

```diff
@@ -0,0 +1,177 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.pds;
+
+  inherit (lib)
+    boolToString
+    getExe
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    optional
+    types
+    ;
+in
+# All defaults are from https://github.com/bluesky-social/pds/blob/8b9fc24cec5f30066b0d0b86d2b0ba3d66c2b532/installer.sh
+{
+  options.services.pds = {
+    enable = mkEnableOption "pds";
+
+    package = mkPackageOption pkgs "pds" { };
+
+    environment = mkOption {
```

Comment:
I'm not sure. This isn't converted to any structured format, whole config is just envvars. Manually defined options below are there because upstream docs doesn't make great job explaining what you actually need to set and these are required

Reply:
Well you could do something like https://github.com/NixOS/nixpkgs/blob/e818264fe227ad8861e0598166cf1417297fdf54/nixos/modules/services/cluster/rke2/default.nix#L255 where the settings names are the env vars and they can be added to the Environment automatically. Just in case there's additional env vars that we don't have that users may want to add. Then they could do the following:

```nix
...
services.pds.settings = {
  PDS_HOSTNAME = "example.com";
  PDS_PORT = 8000;
  PDS_DID_PLC_URL = "https://plc-mirror.example.com";
};
...
```
etc. While the names are uglier, this has precedent, see vaultwarden's freeform `config`(it should be `settings` but sadly, classic nixpkgs moment) allowing you to set things as either camelCase or UPPER_SNAKE_CASE but it autoconverts them at settings generation time(see the vaultwarden module for the conversion code there, it also uses env vars.

Just keeping things consistant like using a freeform `settings` option as defined by RFC 42 when possible is a good idea in my book as it reduces the already high mental energy needed to use Nix/NixOS.